apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/match/ProtectiveShieldMatcher.java

```diff
@@ -46,7 +46,7 @@ public boolean matches(T target) {
         try {
             return this.matcher.matches(target);
         } catch (Throwable t) {
-            logger.warn(t, "Byte-buddy occurs exception when match type.");
+            logger.debug("Byte-buddy occurs exception when match type: {}", t.getMessage());
```

Comment:
Maybe it is better to use `if logger.debugenable` then `logger.warn`? The debug log like this will miss the error stack.

Reply:
> Maybe it is better to use `if logger.debugenable` then `logger.warn`? The debug log like this will miss the error stack.

What I meant is to eliminate the stack traces, they look overwhelming and make users think that the agent is not working well

And the log level is `DEBUG` by default, `if (logger.isDebugEnable) then logger.warn` behaves the same as before

Reply:
I just have concerns about no stack when we really need this.

Reply:
> I just have concerns about no stack when we really need this.

I had looked carefully into the stack traces when I wanted to find out why it really failed, and I'm sure the stack traces don't help at all.

So what about overloading a `debug` method with `Throwable`, and making the log level default to `INFO`?

Reply:
I mean:

```java
if (logger.isDebugEnabled()) {
    logger.debug(throwable, "xxx")
}
```